src/mlpack/bindings/cli/parse_command_line.hpp

```diff
@@ -29,127 +31,64 @@ PARAM_FLAG("verbose", "Display informational messages and the full list of "
 PARAM_FLAG("version", "Display the version of mlpack.", "V");
 
 /**
- * Parse the command line, setting all of the options inside of the CLI object
+ * Parse the command line, setting all of the options inside of the CMD object
  * to their appropriate given values.
  */
 void ParseCommandLine(int argc, char** argv)
 {
   // First, we need to build the boost::program_options variables for parsing.
-  using namespace boost::program_options;
-  options_description desc;
-  variables_map vmap;
+  CLI::App app;
+  app.set_help_flag();
 
   // Go through list of options in order to add them.
-  std::map<std::string, util::ParamData>& parameters = CLI::Parameters();
-  typedef std::map<std::string, util::ParamData>::const_iterator IteratorType;
-  std::map<std::string, std::string> boostNameMap;
-  for (IteratorType it = parameters.begin(); it != parameters.end(); ++it)
+  std::map<std::string, util::ParamData>& parameters = CMD::Parameters();
+  using ItType = std::map<std::string, util::ParamData>::iterator;
+
+  for (ItType it = parameters.begin(); it != parameters.end(); ++it)
   {
     // Add the parameter to desc.
-    const util::ParamData& d = it->second;
-    CLI::GetSingleton().functionMap[d.tname]["AddToPO"](d, NULL,
-        (void*) &desc);
-
-    // Generate the name the user passes on the command line.
-    std::string boostName;
-    CLI::GetSingleton().functionMap[d.tname]["MapParameterName"](d, NULL,
-        (void*) &boostName);
-    boostNameMap[boostName] = d.name;
+    util::ParamData& d = it->second;
+    CMD::GetSingleton().functionMap[d.tname]["AddToPO"](d, NULL,
+        (void*) &app);
```

Comment:
```suggestion
    CMD::GetSingleton().functionMap[d.tname]["AddToPO"](d, NULL, (void*) &app);
```
It fits on one line. :)